src/tests/Common/CLRTest.Execute.Bash.targets

```diff
@@ -272,7 +272,22 @@ $__Command msbuild $CORE_ROOT/wasm-test-runner/WasmTestRunner.proj /p:NetCoreApp
    ]]>
       </BashCLRTestPreCommands>
 
-      <BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetArchitecture) != 'wasm' ">
+      <BashCLRTestPreCommands Condition="'$(TargetOS)' == 'Android'">
+   <![CDATA[
+# Build Android app containing the test dll
+__Command=""
+if [ ! -z ${__TestDotNetCmd+x} ] %3B then
+  __Command+=" $__TestDotNetCmd"
+else
+  __Command+=" dotnet"
+fi
+
+$__Command publish $CORE_ROOT/android-test-runner/AndroidTestRunner.proj -c $(Configuration) -r android-$(TargetArchitecture) /p:Platform=$(TargetArchitecture) /p:TargetArchitecture=$(TargetArchitecture) /p:NetCoreAppCurrent=$(NetCoreAppCurrent) /p:TestAssembly=`pwd`/$(MsBuildProjectName).dll /p:TestBinDir=`pwd`
+
+   ]]>
+      </BashCLRTestPreCommands>
+
+      <BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetArchitecture) != 'wasm' And '$(TargetOS)' != 'Android'">
```

Comment:
Isn't ` And $(TargetArchitecture) != 'wasm' And '$(TargetOS)' != 'Android'` redundant since the property is overridden below by 
`<BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetArchitecture) == 'wasm' ">`
and 
`<BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetOS) == 'Android' ">`

Reply:
They are mutually exclusive. Those are 3 different scenarios. None of them couldn't be overwritten by any other two scenarios.